Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -168,6 +178,89 @@ public final class Chat {
         try await channelUpdater.removeMembers(currentUserId: currentUserId, cid: cid, userIds: Set(members), message: systemMessage)
     }
     
+    /// Loads an array of members for the specified query.
+    ///
+    /// - Parameters:
+    ///   - filter: The filter to use for fetching channel members.
+    ///   - sort: An array of sorting parameters.
+    ///   - pagination: The pagination configuration which includes a limit and an offset or a cursor.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: An array of paginated channel members for the query.
+    public func loadMembers(with filter: Filter<MemberListFilterScope>, sort: [Sorting<ChannelMemberListSortingKey>] = [], pagination: Pagination) async throws -> [ChatChannelMember] {
+        var query = ChannelMemberListQuery(cid: cid, filter: filter, sort: sort)
+        query.pagination = pagination
+        return try await memberListUpdater.load(query)
+    }
+    
+    /// Returns an observable member list state for the specified query.
+    ///
+    /// Calling ``Chat.loadMembers(with:sort:pagination:)`` automatically updates ``MemberListState.members`` for the same query.
+    ///
+    /// - Parameters:
+    ///   - filter: The filter to use for fetching channel members.
+    ///   - sort: An array of sorting parameters.
+    ///   - pagination: The pagination configuration which includes a limit and an offset or a cursor.
+    ///
+    /// - Returns: An instance of `MemberListState` which conforms to the `ObservableObject`.
+    public func makeMemberListState(for filter: Filter<MemberListFilterScope>, sort: [Sorting<ChannelMemberListSortingKey>] = [], pagination: Pagination) async throws -> MemberListState {
```

Comment:
Another idea: why don't we simplify even further - this can be a member list array in the chat state. It also fits conceptually, removes some boilerplate code and it's actually easier for customers. Wdyt?

Reply:
@martinmitrevski I don't think that will work because you might have multiple members list, and not just one. What I think it makes sense is to move this to the `ChatClient` level, and create the member list state there. AFAIK you could create a member list query here that has nothing to do with this Chat. Since the customer can query a member list of any channel, and not only this one, I think it makes sense to move this one level higher

Reply:
Those are good points, but we need to think this through a bit more. I mean the same thing could be said about e.g. watchers as well - you might want to access them without creating a chat. But on the other hand, that will put too much responsibility to the client. 

The question is - how expensive would it be to create a chat, since naturally/logically the members should be part of the chat.
The other question - how frequently would customers want to do that? If we optimize for a small chunk, it's probably not worth it.

Reply:
I'll think about all the things said here. It kind of sounds nice if `ChatState` has `members` but at the same time there is still a way to create member lists by not creating the Chat instance.

Reply:
This is implemented